scripts/asm-installer/install_asm

```diff
@@ -472,6 +523,60 @@ EOF
   WORKLOAD_POOL="${PROJECT_ID}.svc.id.goog"
 }
 
+validate_certs() {
+  if [[ "${CA}" != "citadel" ]]; then
+    fatal "You must select Citadel as the CA in order to use custom certificates."
+  fi
+  if [[ -z "${CA_ROOT}" || -z "${CA_KEY}" || -z "${CA_CHAIN}" || -z "${CA_CERT}" ]]; then
+    fatal "All four certificate options must be present to use a custom cert."
+  fi
+  while read -r CERT_FILE; do
+    if ! [[ -f "${!CERT_FILE}" ]]; then
+      fatal "Couldn't find file ${!CERT_FILE}."
+    fi
+  done <<EOF
+CA_CERT
+CA_ROOT
+CA_KEY
+CA_CHAIN
+EOF
+
+  CA_CERT="$(readlink -f "${CA_CERT}")"; readonly CA_CERT;
+  CA_KEY="$(readlink -f "${CA_KEY}")"; readonly CA_KEY;
+  CA_CHAIN="$(readlink -f "${CA_CHAIN}")"; readonly CA_CHAIN;
+  CA_ROOT="$(readlink -f "${CA_ROOT}")"; readonly CA_ROOT;
+
+  info "Checking certificate files for consistency..."
+  if ! openssl rsa -in "${CA_KEY}" -check >/dev/null 2>/dev/null; then
+    fatal "${CA_KEY} failed an openssl consistency check."
+  fi
+  if ! openssl x509 -in "${CA_CERT}" -text -noout >/dev/null; then
+    fatal "${CA_CERT} failed an openssl consistency check."
+  fi
+  if ! openssl x509 -in "${CA_CHAIN}" -text -noout >/dev/null; then
```

Comment:
I think openssl x509 only checks a single one cert in the PEM file. But it's OK here because you have the cert chain verification later.